src/models/version.rs

```diff
@@ -43,6 +43,9 @@ pub struct NewVersion {
 pub struct TopVersions {
     /// The "highest" version in terms of semver
     pub highest: Option<semver::Version>,
+    /// The "highest" non-prerelease version, or, if only
+    /// prereleases exist, the "highest" version
```

Comment:
I was assuming the semantics would be that if there are only prerelease versions (or all stable versions are yanked), then this would be `None`.

Reply:
as I said, naming is hard :D

this matches the business logic that we are currently doing on the frontend when visiting a crate and figuring out which version to display. it is essentially the "most relevant version".

does that make sense?

Reply:
Should we keep that bit of business logic in the frontend? Using `None`/`null` here to mean "no stable releases" (either never, or all since yanked) seems a bit more useful than providing a fallback value in the backend.

We don't need that distinction in the frontend currently, but it seems nice to provide it by the backend anyway.

Reply:
the problem is that we can't perform that logic everywhere on the frontend because e.g. for search results it would cause N+1 queries since the results only include the crate, but not the versions. for the crate details page this is acceptable but not for crate lists. I'd prefer to use the same logic everywhere to stay consistent.

we could also rename it to default_version or something like that if it helps?

Reply:
I don't follow why this would be N+1 on the frontend. The search API returns values for `max_version`, `newest_version`, and now `max_stable_version`. Can't the frontend fallback to `max_version` if there is no `max_stable_version`? I don't think search results would require additional queries.

Reply:
yeah, I guess with using both fields this could work. my sidegoal was to eventually get rid of the other two fields, but I don't have strong feelings on that.

I'll change it to remove the fallback.

Reply:
> my sidegoal was to eventually get rid of the other two fields, but I don't have strong feelings on that.

We should keep them in case there are third party consumers. (Long term, we should at least start planning a v2 API and tracking our known v1 warts.)

> I'll change it to remove the fallback.

Thanks! Would you update the doc comment as well?

Reply:
> Would you update the doc comment as well?

whoops, sorry about that. done :)